### DISCUSSION TITLE
Custom Grammars - Where to Start

### DISCUSSION THREAD

**Discussion by jamsoft** [danipen/TextMateSharp#21] [Q&A] [answered]
I'm using AvaloniaEdit and am using it as a replacement for a WPF RichTextBox I was using in a custom control.

Look at the AE docs it says I need to implement IRegistryOptions but I can't find any examples of what any of the requirements are for doing so. This is all pretty complex stuff and I can't find anything to give me a nudge in the right direction.

I found a message saying to look at the implementation for AvaloniaEdit but the link is now a 404 and looking over all the branches it seems to have vanished.

My requirements are really simple.

I just have a handful of key words to be green and a handful of keywords to be Red but I just cannot figure out what to do with the interface. It all seems geared up for a much more complex use case and I'm left struggling a bit to be honest.

Are there any docs I'm missing or examples?

**Comment by danipen** [comment, marked as answer]
I don't think TextmateSharp is the piece you should use for that. I'd use AvaloniaEdit's transformers for that. 

Basically you need to write a line transformer that just change the chunk style when you find a match in your line.

You can find a custom transformer here: https://github.com/AvaloniaUI/AvaloniaEdit/blob/master/src/AvaloniaEdit.TextMate/GenericLineTransformer.cs

**Reply by jamsoft** [reply]
Wow, thank you for the prompt reply.

From just a brief look at the example this is incredibly reminiscent of what I was doing with Word Ranges and Runs in the RichTextBox.

Excellent. Thanks so much, off to have a play.

**Comment by jamsoft** [comment]
Wow, took a few hours of reading code and tinkering but it's all working dream-like now. Many thanks for your excellent control and help.

**Reply by zflemingg1** [reply]
Any chance you have a poc I could take a look at. My requirements are similar and setting up a grammar seems like overkill. And can't find any documentation on where to start
